### DISCUSSION TITLE
Download Historical Certificates

### DISCUSSION THREAD

**Discussion by tonyjameshart** [Venafi/venafips#265] [Ideas]
Is there a way to download or "restore" the last historical certificate using VenafiPS?
The only way I know how to have what I want, which is: When a renewal happens, the historical (previous) certificate is shown as a cert object in the same path as the "renewed" cert. I assume you can use a workflow that makes the API calls required to download the previous version cert as a file, then import that cert with a different nickname (add "-previous" into the same location. I hope there is an easier way than to use API calls or manually done via GUI.

**Comment by tonyjameshart** [comment]
It would be great to have 
export-vdccertificate -history 1 
the number would be which historical to export.
or 
export-vdccertificate -vaultID

**Comment by gdbarron** [comment]
Hi @tonyjameshart.

To get the historical certificate data you can use `Get-VdcCertificate -IncludePreviousVersions`.  The return object will have a property named PreviousVersions with CertificateDetails and VaultId.  Check ValidTo and ValidFrom within CertificateDetails to determine which certificate you would like and grab the associated VaultId.

To retrieve the certificate data from the VaultId, use https://docs.venafi.com/Docs/currentSDK/TopNav/Content/SDK/WebSDK/r-SDK-POST-Secretstore-retrieve.php.  VenafiPS has a function to make it easy to use APIs which haven't been integrated to the module, eg. `Invoke-VenafiRestMethod -UriLeaf 'SecretStore/Retrieve' -Body @{'VaultID' = $cert.PreviousVersions[0].VaultId}`.
